PD-121 update directory services articles (#1759)
truenas/documentation: content/SCALE/SCALETutorials/Credentials/DirectoryServices/ConfigIDmapSCALE.md

---
title: "Configuring IDMap"
description: "This article provides instructions on configuring and managing ID mapping in SCALE."
weight: 35
aliases:
tags:
- scaleidmap
---

{{< toc >}}

{{< hint danger >}}
Only administrators experienced with configuring Id mapping should attempt to add new or edit existing idmaps. 

> Misconfiguration can impact system operation.

Reviewer: Idmap settings exist for the purpose of integration with an existing active directory domain to ensure that UIDs and GIDs assigned to active directory users and groups have consistent values domain-wide. The correct configuration therefore relies on details that are entirely external to the truenas server, e.g. how the AD administrator has configured other Unix-like computers in the environment. 

The default is to use an algorithmic method of generating IDs based on the RID component of the user or group's SID in active directory.

Reviewer: Misconfiguration may lead to permissions being incorrectly assigned to users or groups in the case where data is transferred to / from external servers via ZFS replication or rsync (or access is being performed via NFS or other protocols that directly access the UIDs / GIDs on files).

---
title: "Configuring IDMap"
description: "This article provides instructions on configuring and managing ID mapping in SCALE."
weight: 35
aliases:
tags:
- scaleidmap
---

{{< toc >}}

> Idmap settings exist for the purpose of integration with an existing directory domain to ensure that UIDs and GIDs assinged to Active Directoery users and groups have consistent values domain-wide. 

Reviewer 2: ```suggestion
Idmap settings exist for the purpose of integration with an existing directory domain to ensure that UIDs and GIDs assigned to Active Directory users and groups have consistent values domain-wide. 
```

> The correct configuration therefor relies on details that are entriely external to the TrueNAS server, e.g., how the AD administrator has configured other Unix-like computers in the environment.

Reviewer 2: ```suggestion
The correct configuration therefore relies on details that are entriely external to the TrueNAS server, e.g., how the AD administrator has configured other Unix-like computers in the environment.
```

> The default is to use an algorithmic method of generating IDs based on the RID component of the uer or group SID in Active Directory.

Reviewer 2: ```suggestion
The default is to use an algorithmic method of generating IDs based on the RID component of the user or group SID in Active Directory.
```

{{< hint warning >}}
Only administrators experienced with configuring Id mapping should attempt to add new or edit existing idmaps. 

> Misconfiguration can lead to permissions incorrectly assigned to users or groups in the case where data is transffered to/from external servers via ZFS replication or rsync (or when access is performed via NFS or other protocols that directly access the UIDs/GIDs on files).

Reviewer 2: ```suggestion
Misconfiguration can lead to permissions incorrectly assigned to users or groups in the case where data is transfered to/from external servers via ZFS replication or rsync (or when access is performed via NFS or other protocols that directly access the UIDs/GIDs on files).
```

{{< /hint >}}

> The **IDdmap** directory service lets users configure and select a backend to map Windows security identifiers (SIDs) to UNIX UIDs and GIDs. Users must enable the **Active Directory** service to configure and use identity mapping (Idmap).

Reviewer 2: ```suggestion
The **Idmap** directory service lets users configure and select a backend to map Windows security identifiers (SIDs) to UNIX UIDs and GIDs. Users must enable the **Active Directory** service to configure and use identity mapping (Idmap).
```
